src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,245 @@
+  [cabreragarciarodriguezpalacios2014]
+* [Hanche-Olsen and Størmer, Jordan Operator Algebras][hancheolsenstormer1984]
+* [Jordan, von Neumann and Wigner, 1934][jordanvonneumannwigner1934]
+* [McCrimmon, A taste of Jordan algebras][mccrimmon2004]
+
+-/
+
+/--
+A (non-commutative) Jordan multiplication.
+-/
+class is_jordan (A : Type*) [has_mul A] :=
+(lmul_comm_rmul : ∀ a b : A, (a * b) * a = a * (b * a))
+(lmul_lmul_comm_lmul: ∀ a b : A, (a * a) * (a * b) = a * ((a * a) * b))
+(lmul_lmul_comm_rmul: ∀ a b : A, (a * a) * (b * a) = ((a * a) * b) * a)
+(lmul_comm_rmul_rmul: ∀ a b : A, (a * b) * (a * a) = a * (b * (a * a)))
+(rmul_comm_rmul_rmul: ∀ a b : A, (b * a) * (a * a) = (b * (a * a)) * a)
+
+/--
+A commutative Jordan multipication
+-/
+class is_comm_jordan (A : Type*) [has_mul A]:=
+(mul_comm: ∀ a b : A, a * b = b * a)
+(jordan: ∀ a b : A, (a * b) * (a * a) = a * (b * (a *a)))
+
+-- A (commutative) Jordan multiplication is also a Jordan multipication
+@[priority 100] -- see Note [lower instance priority]
+instance is_comm_jordan.to_is_jordan (A : Type*) [has_mul A] [is_comm_jordan A] : is_jordan A :=
+{ lmul_comm_rmul := λ a b, by rw [is_comm_jordan.mul_comm, is_comm_jordan.mul_comm a b],
+  lmul_lmul_comm_lmul := λ a b, by rw [is_comm_jordan.mul_comm (a * a) (a * b),
+  is_comm_jordan.jordan, is_comm_jordan.mul_comm b (a * a)],
+  lmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm, ←is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm],
+  lmul_lmul_comm_rmul :=  λ a b, by rw [is_comm_jordan.mul_comm (a * a) (b * a),
+    is_comm_jordan.mul_comm b a, is_comm_jordan.jordan, is_comm_jordan.mul_comm,
+    is_comm_jordan.mul_comm b (a * a)],
+  rmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm b a, is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm], }
+
+universe u
+
+/- A (unital, associative) ring satisfies the (non-commutative) Jordan axioms-/
+@[priority 100] -- see Note [lower instance priority]
+instance semigroup.is_jordan (B : Type u) [semigroup B] : is_jordan B :=
+{ lmul_comm_rmul := by { intros, rw mul_assoc },
+  lmul_lmul_comm_lmul := by { intros, rw [mul_assoc, mul_assoc] },
+  lmul_comm_rmul_rmul := by { intros, rw [mul_assoc] },
+  lmul_lmul_comm_rmul := by { intros, rw [←mul_assoc] },
+  rmul_comm_rmul_rmul := by { intros, rw [← mul_assoc, ← mul_assoc] } }
+
+variables {A : Type*} [non_unital_non_assoc_ring A]
+
+/--
+Left multiplication operator
+-/
+@[simps] def function.End.L : A →+ add_monoid.End A := add_monoid_hom.mul
+local notation `L` := function.End.L
+
+/--
+Right multiplication operator
+-/
+@[simps] def function.End.R : A→+(add_monoid.End A) :=
+  add_monoid_hom.flip (L  : A →+ add_monoid.End A)
+local notation `R` := function.End.R
+
+-- The Jordan axioms can be expressed in terms of commuting multiplication operators
+
+lemma lmul_rmul_comm [is_jordan A] (a : A) : ⁅L a, R a⁆ = 0 :=
+begin
+  ext b,
+  rw ring.lie_def,
```

Comment:
`apply sub_eq_zero_of_eq` should clean this up

Reply:
Done, thanks.

Reply:
I had a much larger cleanup in mind - see the commit I pushed.

Reply:
Thanks for all your help!